libs/multicall/src/multicall.ts

```diff
@@ -0,0 +1,39 @@
+import type { Multicall3 } from '@cowprotocol/abis'
+import type { Web3Provider } from '@ethersproject/providers'
+
+import { DEFAULT_BATCH_SIZE } from './const'
+import { getMulticallContract } from './utils/getMulticallContract'
+
+export interface MulticallOptions {
+  batchSize?: number
+}
+
+/**
+ * TODO: return results just after batch execution
+ * TODO: add fallback for failed calls
+ * TODO: add providers fallback
```

Comment:
Not yet. I would implement it in the next iteration.
This multicall works more sustainable that uniswap, because:
 - we use `tryAggregate` that means we don't fail if only of calls is reverted
 - we split calls into batches that prevents high load on RPC nodes
 
 But anyway, we should implement those TODOs